osu.Server.Spectator/Hubs/MultiplayerQueue.cs

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System.Linq;
+using System.Threading.Tasks;
+using osu.Game.Online.Multiplayer;
+using osu.Game.Online.Multiplayer.Queueing;
+using osu.Game.Online.Rooms;
+using osu.Game.Rulesets;
+using osu.Server.Spectator.Database;
+using osu.Server.Spectator.Database.Models;
+using osu.Server.Spectator.Extensions;
+
+namespace osu.Server.Spectator.Hubs
+{
+    public class MultiplayerQueue
+    {
+        public const int PER_USER_LIMIT = 3;
+
+        public MultiplayerPlaylistItem CurrentItem => room.Playlist[currentIndex];
+
+        private readonly ServerMultiplayerRoom room;
+        private readonly IMultiplayerServerMatchCallbacks hub;
+
+        private IDatabaseFactory dbFactory = null!;
```

Comment:
Would you be against making this nullable and adding null check / throw on any method which requires it? Feels safer (ensures that `Initialise` is called).